src/rgw/driver/rados/rgw_sync_module_es.cc

```diff
@@ -506,10 +506,8 @@ struct es_obj_metadata {
           const ACLGrant& grant = acliter.second;
           if (grant.get_type().get_type() == ACL_TYPE_CANON_USER &&
               ((uint32_t)grant.get_permission().get_permissions() & RGW_PERM_READ) != 0) {
-            rgw_user user;
-            if (grant.get_id(user)) {
-              permissions.insert(user.to_str());
-            }
+            const auto user = grant.get_user();
+            permissions.insert(user->id.to_str());
```

Comment:
what if `user` is nullptr?

Reply:
this block is conditional on `grant.get_type().get_type() == ACL_TYPE_CANON_USER`, so it won't be null

but it's probably better to check for null there directly, instead of inferring it from ACL_TYPE_CANON_USER